packages/web3-errors/src/errors/transaction_errors.ts

```diff
@@ -347,7 +347,7 @@ export class TransactionGasMismatchError extends InvalidValueError {
 			}, maxPriorityFeePerGas: ${value.maxPriorityFeePerGas ?? 'undefined'}, maxFeePerGas: ${
 				value.maxFeePerGas ?? 'undefined'
 			}`,
-			'transaction must specify legacy or fee market gas properties, not both',
+			'Transaction must define "gas" and "gasPrice" for type 0 transactions or "gas", "maxPriorityFeePerGas" and "maxFeePerGas" for type 2 transactions, not both',
```

Comment:
Along with last one, this will also be breaking change as compared to current 4.x

Reply:
Can we add some inner error ( for more details ) ? so we dnt break API, and add more details in that inner error. or do you want to propose any other way of doing it without breaking changes?